src/libraries/System.Net.Sockets/tests/FunctionalTests/Connected.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class Connected
+    {
+        [Fact]
+        public void NonBlockingFailedConnect_ConnectedReturnsFalse()
+        {
+            using Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            socket.Blocking = false;
+
+            // Connect to port 1 where we expect no server to be listening.
+            SocketException se = Assert.ThrowsAny<SocketException>(() => socket.Connect(IPAddress.Loopback, 1));
+
+            Assert.Equal(SocketError.WouldBlock, se.SocketErrorCode);
+
+            // Give the non-blocking connect some time to complete.
+            socket.Poll(5_000, SelectMode.SelectWrite);
```

Comment:
Seems like this could end up being flaky, especially in CI?

Reply:
I assumed this timeout was milliseconds, but it's microseconds. I've increased the value so the timeout is 5 seconds.

If the connect doesn't complete within this time, the test still passes (`IsConnected` is `false`).

Now, I wonder whether 5 seconds will actually need to pass on Windows. From the docs I learned that on Windows, the Socket doesn't become writable but flags error instead. I'll be able to see the actual timeout from the CI test duration.

Reply:
> I wonder whether 5 seconds will actually need to pass on Windows. From the docs I learned that on Windows, the Socket doesn't become writable but flags error instead. I'll be able to see the actual timeout from the CI test duration.

CI duration for this test on various Windows configurations is less than 20ms.